src/panels/profile/ha-dialog-change-password.js

```diff
@@ -0,0 +1,152 @@
+import '@polymer/paper-button/paper-button.js';
+import '@polymer/paper-dialog/paper-dialog.js';
+import '@polymer/paper-spinner/paper-spinner.js';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import '../../resources/ha-style.js';
+
+import LocalizeMixin from '../../mixins/localize-mixin.js';
+
+/*
+ * @appliesMixin LocalizeMixin
+ */
+class HaDialogChangePassword extends LocalizeMixin(PolymerElement) {
+  static get template() {
+    return html`
+    <style include="ha-style-dialog">
+      .error {
+        color: red;
+      }
+      paper-dialog {
+        max-width: 500px;
+      }
+      .username {
+        margin-top: -8px;
+      }
+    </style>
+    <paper-dialog id="dialog" with-backdrop opened="{{_opened}}" on-opened-changed="_openedChanged">
+      <h2>Change Password</h2>
+      <div>
+        <template is="dom-if" if="[[_errorMsg]]">
+          <div class='error'>[[_errorMsg]]</div>
+        </template>
+        <paper-input
+          autofocus
+          class='currentPassword'
+          label='Current Password'
+          type='password'
+          value='{{_currentPassword}}'
+          required
+          auto-validate
+          error-message='Required'
+        ></paper-input>
+        <paper-input
+          label='New Password'
+          type='password'
+          value='{{_newPassword}}'
+          required
+          auto-validate
+          error-message='Required'
+        ></paper-input>
```

Comment:
We should ask user input new password twice, or allow user unmask the password, likes

https://github.com/home-assistant/home-assistant-polymer/blob/3f0824bd396478120602f8158f53fbcd6d5ae410/src/components/ha-form.js#L40-L55